### ISSUE TITLE
"Failed to create instance" : remote WAD session creation keeps failing

### ISSUE THREAD

**Issue by pradeipp** [microsoft/WinAppDriver#1522]
Whenever I try to create a remote WinAppDriver Session from the "root" app(desktop), I get this error: 
`Failed to create instance`

Here is the session call:

```csharp
var options = new AppiumOptions();
options.AddAdditionalCapability("app", "Root");
options.AddAdditionalCapability("ms:experimental-webdriver", true);
var desktopSession = new WindowsDriver<WindowsElement>(new Uri("http://173.32.11.43:4723/wd/hub"), options, TimeSpan.FromSeconds(120));
```

and here is WAD server log:

```
POST /wd/hub/session HTTP/1.1
Accept: application/json, image/png
Connection: Keep-Alive
Content-Length: 215
Content-Type: application/json;charset=utf-8
Host: 173.32.11.43:4723
User-Agent: selenium/3.141.0 (.net windows)

{"desiredCapabilities":{"app":"Root","ms:experimental-webdriver":true,"platformName":"Windows"},"capabilities":{"firstMatch":[{"ms:experimental-webdriver":true,"platformName":"Windows"}]}}
SessionManager - ERROR : WinAppDriver loading MitaBroker : IRequestHandlePtr::CreateInstance failed w/err 0x800401f0
HTTP/1.1 500 Internal Error
Content-Length: 85
Content-Type: application/json

{"status":13,"value":{"error":"unknown error","message":"Failed to create instance"}}
```

**Comment by anunay1** [comment]
You ate missing http

**Comment by pradeipp** [comment]
@anunay1  sorry about that, I manually typed a random ip address there. My actual script has the proper url. I'll update it above.

**Comment by anunay1** [comment]
How are you starting the WAD server?

**Comment by pradeipp** [comment]
I tried to open the WAD server via both powershell and cmd in admin mode, in 2 different machines. I start it the same way as always: `C:\FOLDER_LOCATION\WinAppDriver.exe REMOTE_IP_ADDR 4723`
I tried both with /wd/hub and without but no luck. I'm not being able to create a remote session at all. I'd thought it might be due to firewall rules but that's also taken care of and its still not working.

**Comment by anunay1** [comment]
Why not use selenium hub?

**Comment by pradeipp** [comment]
I use both web and desktop automation in tandem for our test cases. Is selenium hub suitable for that? would you mind pointing me to a good resource to set it up quickly

**Comment by anunay1** [comment]
That should not be a problem
Check
https://github.com/microsoft/WinAppDriver/blob/master/Docs/SeleniumGrid.md

**Comment by anunay1** [comment]
I don't understand what are you trying to achieve
It will be more clear if you can explain the use case

**Comment by pradeipp** [comment]
So the use case is: we have an application whose workflows include opening word documents from the UI, performing some word macro actions, and then going back to web. So this whole workflow uses both selenium webdriver and winappdriver. The scripts used to work fine but since a few days I keep getting the above error. I tried everything, tried to trace back everything but nothing lets me create a remote session. 
I can create a local winappdriver session and perform the workflows perfectly fine though.

**Comment by shoaibmansoor** [comment]
Hi @pradeipp Some quick checks
1. Have you followed everything from [here](https://github.com/microsoft/WinAppDriver/blob/master/Docs/RunningOnRemoteMachine.md) ?
2. Is developer mode enabled?
3. What is the behavior on removing `ms:experimental-webdriver`?
4. You are using AppiumOptions then why not use appium to control the Winappdriver?
you can install appium by: `npm i -g appium@1.17.1`

If nothing works, consider providing complete winappdriver logs

**Comment by anunay1** [comment]
Both the remote PC and the solution PC on the same subnet, if not then it will not work.

**Comment by pradeipp** [comment]
@shoaibmansoor Yeah all the things you listed, I'd already done it. The setup has worked fine for the past 9months or so. This error came out of nowhere and now it's resolved on its own. 
@anunay1 I was about to try that but it got resolved on its own somehow. 

I didn't change anything, so I'm guessing it's either some organization controlled internal network config thing that caused it for a little while and got resolved over the weekend, or could be the stale state of the machines(which is unlikely because I did restart both machines and that didn't help). Would've liked to find the cause but looks like I'm gonna have to wait until the next time it happens(hope it doesn't). 
I'll leave the issue open for we don't know what caused it or how to resolve it.

**Comment by pradeipp** [comment]
For anyone facing this issue in the future: The winappdriver server in my remote machine was giving connection issues so I wanted to do a health check by pinging it's api directly from my c# code. Maybe some header in my requests was messed, whenever I made direct api calls to the WAD server, this issue was almost always there. I stopped doing that and created a small python project to call it using the winappdriver package instead and now the problem is gone. Hope this helps.
